src/main/java/com/linkedin/kmf/services/ClusterTopicManipulationService.java

```diff
@@ -0,0 +1,242 @@
+ * file except in compliance with the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.linkedin.kmf.services;
+
+import com.linkedin.kmf.XinfraMonitorConstants;
+import com.linkedin.kmf.services.metrics.ClusterTopicManipulationMetrics;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.CreateTopicsResult;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.requests.DescribeLogDirsResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Service monitoring the creations and deletions of Kafka Cluster's Topic.
+ */
+public class ClusterTopicManipulationService implements Service {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ClusterTopicManipulationService.class);
+  private final String _configDefinedServiceName;
+  private final int _reportIntervalSecond;
+  private final ScheduledExecutorService _executor;
+  private final AdminClient _adminClient;
+  private volatile boolean _isOngoingTopicCreationDone;
+  private String _currentlyOngoingTopic;
+  private CreateTopicsResult _createTopicsResult;
+  private AtomicInteger _totalPartitions = new AtomicInteger();
+  // TODO -- ClusterTopicManipulationMetrics implementation in progress!
+  private ClusterTopicManipulationMetrics _clusterTopicManipulationMetrics;
+
+  public ClusterTopicManipulationService(String name, AdminClient adminClient) {
+    LOGGER.info("ClusterTopicManipulationService constructor initiated {}", this.getClass().getName());
+
+    _isOngoingTopicCreationDone = true;
+    _adminClient = adminClient;
+    _executor = Executors.newSingleThreadScheduledExecutor();
+    _reportIntervalSecond = 1;
+    _configDefinedServiceName = name;
+    // TODO: instantiate a new instance of ClusterTopicManipulationMetrics(..) here.
+  }
+
+  /**
+   * The start logic must only execute once.  If an error occurs then the implementer of this class must assume that
+   * stop() will be called to clean up.  This method must be thread safe and must assume that stop() may be called
+   * concurrently. This can happen if the monitoring application's life cycle is being managed by a container.  Start
```

Comment:
It seems to me that the `start()` and `stop()` methods are not thread-safe and there is no guarantee that the `start logic must only execute once`. I would add two things.

1. Thread-safety to these 2 methods

2. Some states to keep track of how many times the `start()` method has been invoked and throw an IllegalStateException if any unexpected invocation is encountered

Reply:
Updated the two methods to behave in a way that's similar to existing ProduceService and ConsumeService. This should suffice